nodeJS/testing/Testing.md

```diff
@@ -151,3 +150,12 @@ If we were using a real database here, then we would want to do something simila
 2. Supertest actually pulls from another related project called Superagent.  Any method that you can call in Superagent you can also call from supertest, so you'll need to take a look through the [SuperAgent docs](http://visionmedia.github.io/superagent/) as well.
 </div>
 
+### Knowledge Checks
+ 
+This section contains questions for you to check your understanding of this lesson. If you’re having trouble answering the questions below on your own, review the material above to find the answer.
+
+- <a class='knowledge-check-link' href='https://github.com/visionmedia/supertest#about'>What is the motivation behind SuperTest?</a>
+- <a class='knowledge-check-link' href='#done'>What is done parameter, where does it come from and how can it be used with SuperTest's .expect() method?</a>
```

Comment:
```suggestion
- <a class='knowledge-check-link' href='#done'>What is done parameter, where does it come from and how can it be used with SuperTest's .expect() method?</a>
```
```suggestion
- <a class='knowledge-check-link' href='#done'>What is the `done` parameter, where does it come from and how can it be used with SuperTest's .expect() method?</a>
```
Minor grammar/style suggestion. Additionally I'm not seeing the answer to "where does it come from and how can it be used with SuperTest's .expect() method?" in the section of our lesson you've linked this question to, can you point it out to me?

Reply:
Your suggestion is great. My intention was for the reader to answer: The `done` parameter can be passed as the test's callback and then called inside .expect() method to signal that the test is complete in case of asynchronous operations - so well basically the linked fragment. You're right that this question could be formulated more clearly, my intention was to make sure that the reader understands the concept as I think it's rather important. Maybe something simpler: What is the purpose of the `done` parameter? But then we'd be repeating the Learning Outcomes

Reply:
@esiale one thing we want to avoid with the Knowledge Checks is asking questions that aren't *directly* answered in our content/linked resources. I would see if you can reformat the question in such a way that the answer is given directly.

If doing so requires that we repeat a Learning Outcome, that may be more of an indication that the Learning Outcome could be reworded than that the Knowledge Check is poorly worded.

Reply:
I've given this some thought and would like to propose moving the learning outcome in question to the knowledge checks. I think it'd be better there. I've reworded the learning outcome. I've committed the changes (this is my first time working with PRs so please don't mind if I've done something wrong)